File: Java/Eta/Source/Stubs/external/com/thomsonreuters/upa/transport/WriteArgs.java
package com.thomsonreuters.upa.transport;

/**
 * UPA Write Arguments used in the {@link Channel#write(TransportBuffer, WriteArgs, Error)} call.
 * 
 * @see Channel
 * @see WritePriorities
 * @see WriteFlags
 */
public interface WriteArgs
{
    /**
     * Priority to flush the message for {@link Channel#write(TransportBuffer, WriteArgs, Error)}.
     * Must be in the range of {@link WritePriorities#HIGH} - {@link WritePriorities#LOW}.
     * 
     * @param priority the priority to set
     * 
     * @see WritePriorities
     */
    public void priority(int priority);

    /**
     * Priority to flush the message for {@link Channel#write(TransportBuffer, WriteArgs, Error)}.
     * 
     * @return the priority
     * 
     * @see WritePriorities
     */
    public int priority();

    /**
     * Write flags used for {@link Channel#write(TransportBuffer, WriteArgs, Error)}.
     * Must be in the range of {@link WriteFlags#NO_FLAGS} - {@link WriteFlags#DIRECT_SOCKET_WRITE}.
     * 
     * @param flags the flags to set
     * 
     * @see WriteFlags
     */
    public void flags(int flags);

    /**
     * Write flags used for {@link Channel#write(TransportBuffer, WriteArgs, Error)}.
     * 
     * @return the flags
     * 
     * @see WriteFlags
     */
    public int flags();

    /**
     * The number of bytes to be written, including any transport header overhead
     * and taking into account any compression, for this write call.
     * 
     * @return the bytesWritten
     */
    public int bytesWritten();

    /**
     * The number of bytes to be written, including any transport header
     * overhead but not taking into account any compression, for this write call.
     * 
     * @return the uncompressedBytesWritten
     */
    public int uncompressedBytesWritten();

    /**
     * Specifies the sequence number of the message {@link Channel#write(TransportBuffer, WriteArgs, Error)}.
     * 
     * @param seqNum the sequence number to set
     */
    public void seqNum(long seqNum);

    /**
     * The sequence number of the message for {@link Channel#write(TransportBuffer, WriteArgs, Error)}.
     * 
     * @return the seqNum
     */
    public long seqNum();
    
    /**
     * Clears UPA Write Arguments.
     */
    public void clear();
}

File: CSharp/Eta/Src/Core/Transport/transport/WriteArgs.cs
/*|-----------------------------------------------------------------------------
 *|            This source code is provided under the Apache 2.0 license
 *|  and is provided AS IS with no warranty or guarantee of fit for purpose.
 *|                See the project's LICENSE.md for details.
 *|           Copyright (C) 2022-2023 LSEG. All rights reserved.     
 *|-----------------------------------------------------------------------------
 */

using LSEG.Eta.Common;
using System.Runtime.CompilerServices;

namespace LSEG.Eta.Transports
{
    /// <summary>
    /// ETA Write Arguments used in the <see cref="IChannel.Write(ITransportBuffer, WriteArgs, out Error)"/> call.
    ///
    /// <seealso cref="IChannel"/>
    /// <seealso cref="WritePriorities"/>
    /// <seealso cref="WriteFlags"/>
    /// </summary>
    sealed public class WriteArgs
    {
        /// <summary>
        /// Priority to flush the message for <see cref="IChannel.Write(ITransportBuffer, WriteArgs, out Error)"/>.
        /// <seealso cref="WritePriorities"/>
        /// </summary>
        /// <vaue>The priority</vaue>
        public WritePriorities Priority { get; set; }

       /// <summary>
       /// Write flags used for <see cref = "IChannel.Write(ITransportBuffer, WriteArgs, out Error)" />.
       /// <seealso cref="WriteFlags"/>
       /// </summary>
       /// <value>The flags</value>
        public WriteFlags Flags { get; set; }

        /// <summary>
        /// The number of bytes to be written, including any transport header overhead
        /// and taking into account any compression, for this write call.
        /// </summary>
        /// <value>The bytes written</value>
        public int BytesWritten { get; internal set; }

        /// <summary>
        /// The number of bytes to be written, including any transport header
        /// overhead but not taking into account any compression, for this write call.
        /// </summary>
        /// <value>The uncompressed bytes written</value>
        public int UncompressedBytesWritten { get; internal set; }

        /// <summary>
        /// The sequence number of the message for <see cref="IChannel.Write(ITransportBuffer, WriteArgs, out Error)"/>.
        /// </summary>
        /// <value>The sequeuce number</value>
        public long SeqNum { get; set; }

        /// <summary>
        /// Clears ETA Write Arguments.
        /// </summary>
        [MethodImpl(MethodImplOptions.AggressiveOptimization | MethodImplOptions.AggressiveInlining)]
        public void Clear()
        {
            Priority = WritePriorities.HIGH;
            Flags = WriteFlags.NO_FLAGS;
            BytesWritten = 0;
            UncompressedBytesWritten = 0;
            SeqNum = 0;
        }

        /// <summary>
        /// Gets the string representation of this object. 
        /// </summary>
        /// <returns>The string value</returns>
        public override string ToString()
        {
            return $"Flags: {Flags}, BytesWritten: {BytesWritten}, SeqNum: {SeqNum}";
        }
    }
}

